File: C++/Linked List/Circular_Linked_List.cpp
#include<iostream>
using namespace std;

class node{
    public:
        int data;
        node *next;
        node(int data){
            this->data = data;
            this->next = nullptr;
        }
};
class circularLinkedList{
    private:
        node* start;
    public:
        circularLinkedList(){
            start = nullptr;
        }

        void Apped(int value){
            node *newNode = new node(value);
            if(!start){
                start = newNode;
                start->next = start;
            }else{
                node *temp = start;
                while(temp->next != start){
                    temp = temp->next;
                }
                temp->next = newNode;
                newNode->next = start;
            }
        }

        void display(){
            node *temp = start;
            do{
                cout<< temp->data<<", ";
                temp = temp->next;
            }while(temp != start);
            cout<<endl;
        }

        void intfirst(int value){
            node *newNode = new node(value);
            if(!start){
                start = newNode;
                start->next = start;
            }else{
                node *temp = start;
                while(temp->next != start){
                    temp = temp->next;
                }
                temp->next = newNode;
                newNode->next = start;
                start = newNode;
            }
        }
};
int main(){

    circularLinkedList cl;
    cl.Apped(58);
    cl.Apped(2);
    cl.Apped(87);
    cl.Apped(11);
    cl.Apped(103);

    cl.intfirst(4);
    cout<<"Your output is: ";
    cl.display();

}

File: Python/Linked_List/Circular_Linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class Circular_Linked_list:
    def __init__(self):
        self.Head = None
    
    def add(self, data):
        new_node = Node(data)
        if self.Head is None:
            self.Head = new_node
            new_node.next = self.Head
            return
        temp = self.Head
        while temp.next != self.Head:
            temp = temp.next
        temp.next = new_node
        new_node.next = self.Head
    
    # other all function are same a normal linked list
    
    def display(self):
        if self.Head is None:
            print("Your Linked list is empty")
            return
        temp = self.Head
        print(temp.data, end=" -> ")
        temp = temp.next
        while temp != self.Head:
            print(temp.data, end=" -> ")
            temp = temp.next
        print(temp.data, end=" -> continue")

if __name__=="__main__":
    l1 = Circular_Linked_list()
    l1.add(85)
    l1.add(4)
    l1.add(7)
    l1.add(3)
    l1.add(78)
    l1.display()
